### PR TITLE
Introduce experiment folder

### PR DESCRIPTION
**Pull request by markusdregi**
This is preparation before addressing #1337.

**Rationale**
When the ert3 workspace was introduced an experiment was denoted as a folder within the workspace. However, this is a massive mistake as it prevents any other folders from existing in the workspace. It could be solved by making files within the folders or similar denote whether a folder represented an experiment, but this would be very hard to navigate by only looking at the file system.

This PR suggests to introduce a folder named `experiments` in the workspace and then conclude that an experiment is described by a folder in `<workspace> / experiments`.

### CODE REVIEW

**Comment by markusdregi** [comment]
This PR nicely displays the current awkward knowledge of the file structure the tool has. This should be resolved when #1340 is resolved.

File: tests/ert3/engine/integration/conftest.py [hunk 1]
```diff
@@ -3,6 +3,8 @@
 import pytest
 import numpy as np
 
+import ert3
```
Comment (sondreso) [inline tests/ert3/engine/integration/conftest.py:6]: Isn't it possible to only import `ert3.workspace` here and noot the entire module? 
```suggestion
import ert3.workspace
```

**Review by sondreso — APPROVED** [review]
Only a small comment, else it looks good 🥼

File: tests/ert3/engine/integration/conftest.py (hunk 1 above)
Comment (markusdregi) [reply tests/ert3/engine/integration/conftest.py]: Good idea 👏

**Comment by sondreso** [comment]
Failing tests in komodo is due to missing dependency for semeio, so think we can merge without that.

**Pull request merged**